src/lib/xmlparse.rs

```diff
@@ -1339,60 +1421,69 @@ pub type KEY = *const XML_Char;
 an attribute has been specified. */
 
 #[repr(C)]
-#[derive(Clone)]
 pub struct ElementType {
     pub name: *const XML_Char,
-    pub prefix: *mut Prefix,
-    pub idAtt: *const AttributeId,
-    pub defaultAtts: Vec<DefaultAttribute>,
+    pub prefix: Cell<Ptr<Prefix>>,
+    pub idAtt: Cell<Ptr<AttributeId>>,
+    pub defaultAtts: RefCell<Vec<DefaultAttribute>>,
 }
 
 impl ElementType {
     fn new() -> Self {
         ElementType {
             name: ptr::null(),
-            prefix: ptr::null_mut(),
-            idAtt: ptr::null(),
-            defaultAtts: Vec::new(),
+            prefix: Cell::new(None),
+            idAtt: Cell::new(None),
+            defaultAtts: RefCell::new(Vec::new()),
         }
     }
 }
 
 #[repr(C)]
-#[derive(Copy, Clone)]
 pub struct DefaultAttribute {
-    pub id: *mut AttributeId,
+    pub id: Rc<AttributeId>,
     pub isCdata: XML_Bool,
     pub value: *const XML_Char,
 }
 
 #[repr(C)]
-#[derive(Copy, Clone)]
 pub struct Entity {
     pub name: *const XML_Char,
-    pub textPtr: *const XML_Char,
-    pub textLen: c_int,
-    pub processed: c_int,
-    pub systemId: *const XML_Char,
-    pub base: *const XML_Char,
-    pub publicId: *const XML_Char,
-    pub notation: *const XML_Char,
-    pub open: XML_Bool,
-    pub is_param: XML_Bool,
-    pub is_internal: XML_Bool,
+    pub textPtr: Cell<*const XML_Char>,
+    pub textLen: Cell<c_int>,
+    pub processed: Cell<c_int>,
+    pub systemId: Cell<*const XML_Char>,
+    pub base: Cell<*const XML_Char>,
+    pub publicId: Cell<*const XML_Char>,
+    pub notation: Cell<*const XML_Char>,
+    pub open: Cell<XML_Bool>,
+    pub is_param: Cell<XML_Bool>,
+    pub is_internal: Cell<XML_Bool>,
```

Comment:
I know that this simplifies a lot, but I keep wondering if it's becoming an anti-pattern if we have to use it absolutely everywhere